api/v1beta1/agentserviceconfig_types.go

```diff
@@ -50,6 +50,13 @@ type MustGatherImage struct {
 	Url string `json:"url"`
 }
 
+type ImageServiceOptions struct {
+	// InsecureSkipVerify toggles whether or not the image service verifies
+	// certificates when downloading iso images from a server.
+	// +optional
+	InsecureSkipVerify string `json:"insecureSkipVerify"`
```

Comment:
It would be better to use a type alias and then define constants with that type for each allowed value. The k8s API conventions [describe that](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types).

Then you can use validation markers similar to this:

```
   // +kubebuilder:validation:Optional
   // +kubebuilder:validation:Enum=true;false
   // +kubebuilder:default:=false
```

This will make it obvious to the API user what values can be used and what the default is, while also taking advantage of built-in validation.

Reply:
The field name makes sense when it's part of a TLS config struct, but may lack context here. Something like `InsecureSkipTLSVerify` would be more clear IMO.